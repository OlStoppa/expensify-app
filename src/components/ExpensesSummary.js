import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';



export const ExpensesSummary = (props) => {
	const length = props.expenses.length;
	const expense = length === 1 ? 'expense' : 'expenses';
	const total = numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00');
	return(
		<div>
			<h1>{`Viewing ${length} ${expense} totalling ${total}`}</h1>
			
		</div>

		);

}

const mapStatetoProps = (state) => {
	return{
		expenses: selectExpenses(state.expenses, state.filters)
		
	};

};

export default connect(mapStatetoProps)(ExpensesSummary);