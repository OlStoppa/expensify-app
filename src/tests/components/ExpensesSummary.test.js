import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with no expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenses={[]} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with an expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
	expect(wrapper).toMatchSnapshot();
});
