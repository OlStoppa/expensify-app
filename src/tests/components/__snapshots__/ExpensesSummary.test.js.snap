// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with an expenses 1`] = `
<div>
  <h1>
    Viewing 1 expense totalling $1.95
  </h1>
</div>
`;

exports[`should render ExpensesSummary with no expenses 1`] = `
<div>
  <h1>
    Viewing 0 expenses totalling $0.00
  </h1>
</div>
`;
